src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -557,15 +587,22 @@ private void ScanInterpolatedStringLiteralHoleBalancedText(char endingChar, bool
 
                             TrySetUnrecoverableError(_lexer.MakeError(_lexer.TextWindow.Position, 1, ErrorCode.ERR_SyntaxError, endingChar.ToString()));
                             goto default;
-                        case '"' when RecoveringFromRunawayLexing():
```

Comment:
I'm pretty sure this would still be emitted as a jump table. Do you have an indication it wouldn't?

Reply:
talked offline.  likely a faulty memory on my part.  that said, i prefer the new pattern, and in raw-interpolated branch we extract out this logic for each specific branch into its own method.